### ISSUE TITLE
[F811] Allow importing fixtures

### ISSUE THREAD

**Issue by alanhdu** [astral-sh/ruff#16602]
### Summary

We have some code where we define `pytest.fixture`s in a centralized place and then import them in our test suite. Something that looks like:

```python
from a import foo

def test_hello(foo) -> None:
    pass
```

Unfortunately, `ruff check` will then fail with:
```
test_b.py:1:15: F401 [*] `a.foo` imported but unused
  |
1 | from a import foo
  |               ^^^ F401
2 |
3 | def test_hello(foo) -> None:
  |
  = help: Remove unused import: `a.foo`

test_b.py:3:16: F811 Redefinition of unused `foo` from line 1
  |
1 | from a import foo
2 |
3 | def test_hello(foo) -> None:
  |                ^^^ F811
4 |     pass
  |
  = help: Remove definition: `foo`
```

Do you know what's going on with the F811 failure? I think the F401 failure is understandable (not sure we want to special-case logic that the fixture is automagically used by pytest, although that would certainly help us out), but I feel like the "redefinition of unused `foo`" error is weird.

**Comment by MichaReiser** [comment]
You can use to https://docs.astral.sh/ruff/settings/#lint_pyflakes_allowed-unused-imports to disable the `F401` warning. Unfortunately, this doesn't suppress F811 which is mainly what you're looking for. Although I'm not sure if ignoring the imports lited in `allowed-unused-imports` is in the spirit of `F811` outside of pytests. Have you considered disabling `F811` in tests?

**Comment by nickurak** [comment]
FWIW, this is common enough that there's a `pylint-pytest`  pylint plugin that specially handles this.

**Comment by nickurak** [comment]
> Do you know what's going on with the F811 failure?

I'm not sure which part of the issue you're not sure about, but just an example of what F811 is trying to prevent: If it was actually something meaningfully imported, then the parameter having that name would obscure it.

```python
import sys
from typing import Any

def non_returning_function(sys):
    sys.exit(0)

class C:
    def exit(self, x):
        print("lol, nvm")

non_returning_function(C())

assert(not "this is unreachable")
```
